### DISCUSSION TITLE
Is it okay to change a plugin's language?

### DISCUSSION THREAD

**Discussion by gabrielcarloto** [Flow-Launcher/Flow.Launcher#2589] [Q&A]
I think the answer may be yes, but just to be sure. I have two plugins: search npm and search mdn. They're both written in typescript and executed with nodejs. Node takes a bit too long to start up on my machine, and I find it annoying to have this little delay between pressing enter and the plugin opening the url.

While there is no progress on #1908 (I think), I rewrote the search npm plugin in go and it opens links much faster. I haven't uploaded it to github yet, but I was wondering if changing the plugin's language and sending an executable without prior review would be nice from a user's perspective and from the flow launcher devs.

**Comment by Garulf** [comment]
Flow Launcher v1.17 introduced a new plugin API called "v2" that allows plugins to continuously run in the background.

The feature is *very* new and not necessarily complete. Several Javascript/TypeScript dev's are actively exploring this API on our Discord.

**Comment by jjw24** [comment]
>I rewrote the search npm plugin in go and it opens links much faster. I haven't uploaded it to github yet, but I was wondering if changing the plugin's language and sending an executable without prior review would be nice from a user's perspective and from the flow launcher devs.

Are you doing this via a new release on your existing plugin in flow?

**Reply by jjw24** [reply]
If so it would be best to update the plugin language info to `executable` in PluginsManifest and we can also have a quick review. I have always wanted to incorporate Go into the supported language list but never had the time to do it, it your plugin is in Go would definately be keen to check it out.
